velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -99,12 +99,53 @@ FOLLY_ALWAYS_INLINE void PrestoHasher::hash<TypeKind::BOOLEAN>(
   });
 }
 
+template <>
+FOLLY_ALWAYS_INLINE void PrestoHasher::hash<TypeKind::BIGINT>(
+    const SelectivityVector& rows,
+    BufferPtr& hashes) {
+  if (vector_->base()->type()->isShortDecimal()) {
+    applyHashFunction(rows, *vector_.get(), hashes, [&](auto row) {
+      return vector_->valueAt<int64_t>(row);
+    });
+  } else {
+    applyHashFunction(rows, *vector_.get(), hashes, [&](auto row) {
+      return hashInteger(vector_->valueAt<int64_t>(row));
+    });
+  }
+}
+
+FOLLY_ALWAYS_INLINE uint64_t mix(uint64_t current, uint64_t value) {
+  return XXH_rotl64(current + value * XXH_PRIME64_2, 31) * XXH_PRIME64_1;
+}
+
+FOLLY_ALWAYS_INLINE uint64_t updateTail(uint64_t hash, uint64_t value) {
+  auto temp = hash ^ mix(0, value);
+  return XXH_rotl64(temp, 27) * XXH_PRIME64_1 + XXH_PRIME64_4;
+}
+
+FOLLY_ALWAYS_INLINE uint64_t hashLongDecimal(const uint64_t value) {
```

Comment:
The name is misleading since LongDecimal's physical type is int128_t and the argument is int64_t.
Maybe `hashLongDecimalPart`?

Reply:
Done